src/java.base/share/classes/java/math/MutableBigInteger.java

```diff
@@ -1892,6 +1913,183 @@ private boolean unsignedLongCompare(long one, long two) {
         return (one+Long.MIN_VALUE) > (two+Long.MIN_VALUE);
     }
 
+    /**
+     * Calculate the integer {@code n}th root {@code floor(nthRoot(this, n))} and the remainder,
+     * where {@code nthRoot(., n)} denotes the mathematical {@code n}th root.
+     * The contents of {@code this} are <em>not</em> changed. The value of {@code this}
+     * is assumed to be non-negative and the root degree {@code n >= 3}.
+     *
+     * @implNote The implementation is based on the material in Richard P. Brent
+     * and Paul Zimmermann, <a href="https://maths-people.anu.edu.au/~brent/pd/mca-cup-0.5.9.pdf">
+     * Modern Computer Arithmetic</a>, p. 27-28.
+     *
+     * @param n the root degree
+     * @return the integer {@code n}th root of {@code this} and the remainder
+     */
+    MutableBigInteger[] nthRootRem(final int n) {
+        // Special cases.
+        if (this.isZero() || this.isOne())
+            return new MutableBigInteger[] { this, new MutableBigInteger() };
+
+        final int bitLength = (int) this.bitLength();
+        // if this < 2^n, result is unity
+        if (bitLength <= n) {
+            MutableBigInteger rem = new MutableBigInteger(this);
+            rem.subtract(ONE);
+            return new MutableBigInteger[] { new MutableBigInteger(1), rem };
+        }
+
+        MutableBigInteger s;
+        if (bitLength <= Long.SIZE) {
+            // Initial estimate is the root of the unsigned long value.
+            final long x = this.toLong();
+            long sLong = (long) Math.ceil(nthRootApprox(Math.nextUp(x >= 0 ? x : x + 0x1p64), n));
```

Comment:
Since the `double` result is truncated to a `long`, meaning it could be an underestimate requiring an initial iteration to produce an overestimate, could it make sense to drop the `ceil()` and just add 1 to the `long`?